manim/utils/updaters.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import annotations
+
+import inspect
+from collections.abc import Callable
+from typing import TYPE_CHECKING, TypeVar, Union
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+    from manim.mobject.mobject import Mobject
+    from manim.mobject.opengl.opengl_mobject import OpenGLMobject
+    from manim.opengl.shader import Object3D
+
+
+__all__ = [
+    "MobjectTimeBasedUpdater",
+    "MobjectNonTimeBasedUpdater",
+    "MobjectUpdater",
+    "MeshTimeBasedUpdater",
+    "MeshNonTimeBasedUpdater",
+    "MeshUpdater",
+    "SceneUpdater",
+    "MobjectUpdaterWrapper",
+    "MeshUpdaterWrapper",
+]
+
+
+M = TypeVar("M", bound=Union["Mobject", "OpenGLMobject"])
+
+MobjectTimeBasedUpdater: TypeAlias = Callable[[M, float], M]
```

Comment:
What do you think about maybe making these shorter, like maybe `MobjectBasicUpdater` and `MobjectDtUpdater`?

Reply:
I like it! I commited your proposal, thanks :)